src/main/groovy/life/qbic/db/MariaDBManager.groovy

```diff
@@ -466,14 +488,22 @@ class MariaDBManager implements IQueryService {
     final String query = "SELECT * from samples WHERE UPPER(id) = UPPER('${sampleId}');"
     try {
       List<GroovyRowResult> results = sql.rows(query)
-      if( results.size() > 0 ) {
+      if (results.size() > 0) {
         GroovyRowResult rs = results.get(0)
         int locationID = rs.get("current_location_ID")
         setStatus(sampleId, locationID, status, sql)
+      } else {
+        throw new NotFoundException("Sample $sampleId could not be found in the database.")
       }
+    } catch (NotFoundException notFoundException) {
+      sql.close()
```

Comment:
is this closed in the "setStatus()" function in the try case? just making sure...

Reply:
setStatus does not close the connection. It ignores exceptions and only prints the stack trace. Do you think we should change that?

Reply:
nevermind, I now notice it is closed in the "finally" case, see suggestions below